src/lib/lib.rs

```diff
@@ -8,7 +8,7 @@ pub(crate) mod circuit_helpers;
 pub mod encoding;
 pub mod services;
 
-#[derive(Debug, Default)]
+#[derive(Debug, Copy, Clone, Default)]
```

Comment:
Is that really needed? When would we need to copy or clone our server instance?

Reply:
Since creating a server takes ownership of the struct passed to it, it needs to be at least copy-able in order to create multiple services. Maybe we can get rid of clone, but for some reason I had an idea in the back of my head that the two are interlinked.